react/features/recording/components/LiveStream/AbstractStartLiveStreamDialog.ts

```diff
@@ -188,13 +217,19 @@ export default class AbstractStartLiveStreamDialog<P extends IProps>
         const { broadcasts, selectedBoundStreamID } = this.state;
         const key
             = (this.state.streamKey || this.props._streamKey || '').trim();
+        const base = (this.state.streamBaseURL || this.props._streamBaseURL || '').trim();
 
+        if (!base) {
+            return false;
+        }
         if (!key) {
             return false;
         }
 
+        const rtmpURL = base.endsWith('/') ? base + key : `${base}/${key}`;
```

Comment:
We should ensure we preserve the functionality of providing a full RTMP URL instead of base + key, for services which may not have the key as the final part of the path.

Reply:
Sure, good point. I made the key field optional, so you can input the full RTMP URL in the first field if needed.